Core/Src/main.c

```diff
@@ -1,181 +1,225 @@
+/* USER CODE BEGIN Includes */
+
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+
+/* USER CODE BEGIN PV */
+#define NUMBER_BYTES 3
+
+/* USER CODE END PV */
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+/* USER CODE BEGIN PFP */
+
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+  /* USER CODE BEGIN 1 */
+
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
+  /* USER CODE BEGIN 2 */
+
+  //Brings gpio pin for high from the initial preset low value
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+  //Tx buffer with start bit, the appropriate control bits for single ended transmission for CH0
+  const uint8_t tx_buffer[NUMBER_BYTES] = {1, 0x80, 0};
+  const uint32_t spi_timeout = 100;
+  uint8_t rx_buffer[NUMBER_BYTES] = {0};
+  uint16_t counter_value;
+  HAL_StatusTypeDef hal_status;
+
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  const double adc_conversion = 3.128;
+
+
+  /* USER CODE END 2 */
+
+  /* Infinite loop */
+  /* USER CODE BEGIN WHILE */
+  // Potentiometer connected to CH0 of ADC
+  while (1)
+  {
+	  //CS line brought low to start data transmission
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	  //Transmitting and Receiving Data
+	  hal_status = HAL_SPI_TransmitReceive(&hspi1, tx_buffer, rx_buffer, NUMBER_BYTES, spi_timeout);
+
+
+	  if(hal_status == HAL_OK)
+	  {
+		  /*Using AND operator to isolate the useful bits stored in the array's second index, and using
+		  bit shifting along with OR operator to combine with the useful bits in the array's third index
+		  followed by converting it to the respective time period value to equal ~10% of clock period; 10%
+		  of 64000*/
+		  counter_value = ((rx_buffer[1] & 3 << 8) | rx_buffer[2]) * adc_conversion + 3200;

```

Comment:
3200 is a magic number, please create a const variable for it!

Reply:
added a const int variable to account for this